File: Coin_Flip_Streak_Tracker.cs
```CSharp
using System;

class Program
{
    static void Main(string[] args)
    {
        Random random = new Random();
        int streak = 0;
        string lastFlip = "";

        while (true)
        {
            int flip = random.Next(0, 2);
            string result = flip == 0 ? "Heads" : "Tails";

            if (lastFlip == "")
            {
                lastFlip = result;
                streak = 1;
            }
            else if (lastFlip == result)
            {
                streak++;
            }
            else
            {
                Console.WriteLine("Streak of " + lastFlip + ": " + streak);
                lastFlip = result;
                streak = 1;
            }

            Console.WriteLine(result);
            System.Threading.Thread.Sleep(1000);
        }
    }
}
```

File: Coin_Flip_Streak_Tracker.java
```java
import java.util.Random;

public class CoinFlipStreakTracker {
    public static void main(String[] args) {
        Random rand = new Random();
        int streak = 0;
        String lastFlip = "";
        for (int i = 0; i < 100; i++) {
            int flip = rand.nextInt(2);
            String currentFlip = flip == 0 ? "Heads" : "Tails";
            System.out.println(currentFlip);
            if (currentFlip.equals(lastFlip)) {
                streak++;
            } else {
                streak = 1;
            }
            System.out.println("Streak: " + streak);
            lastFlip = currentFlip;
        }
    }
}
```